Description: Find llvm-3.5 include files
 llvm-3.5-dev has changed the location for an header file, leading
 to a ftbfs.

Author: Gianfranco Costamagna <costamagnagianfranco@yahoo.it>
Origin: debian
Bug-Debian: http://bugs.debian.org/763238
Forwarded: https://code.google.com/p/gambas/issues/detail?id=564
Last-Update: 2014-09-29

--- a/gb.jit/configure.ac
+++ b/gb.jit/configure.ac
@@ -25,7 +25,7 @@
 
 GB_COMPONENT(
   jit, JIT, gb.jit, [src],
-  [GB_FIND(llvm/AutoUpgrade.h llvm-c/Core.h, /usr/lib/llvm* /usr/local/lib /usr/local /usr/lib /usr `$LLVM_CONFIG --prefix`, include)],
+  [GB_FIND(llvm/Config/llvm-config.h llvm-c/Core.h, /usr/lib/llvm* /usr/local/lib /usr/local /usr/lib /usr `$LLVM_CONFIG --prefix`, include)],
   [GB_FIND(libLLVM-$LLVM_VERSION.$SHLIBEXT, /usr/lib/llvm* /usr/local /usr `$LLVM_CONFIG --prefix`, lib)],
   [$C_LIB `$LLVM_CONFIG --ldflags` -lLLVM-$LLVM_VERSION],
   [-I../../main/gbx -I../../main/share -D_DEBUG -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS])
--- a/gb.jit/src/jit.h
+++ b/gb.jit/src/jit.h
@@ -42,7 +42,13 @@
 #include "llvm/Intrinsics.h"
 #include "llvm/Instructions.h"
 #endif
+#if (LLVM_VERSION_MAJOR == 3 && LLVM_VERSION_MINOR >= 5)
+#include "llvm/IR/Verifier.h"
+#include "llvm/IR/CFG.h"
+#else
 #include "llvm/Analysis/Verifier.h"
+#include "llvm/Support/CFG.h"
+#endif
 #include "llvm/ExecutionEngine/JIT.h"
 #include "llvm/ExecutionEngine/Interpreter.h"
 #include "llvm/ExecutionEngine/GenericValue.h"
@@ -57,7 +63,6 @@
 #include "llvm/Support/IRBuilder.h"
 #endif
 #include "llvm/Support/DynamicLibrary.h"
-#include "llvm/Support/CFG.h"
 #include "llvm/PassManager.h"
 #include "llvm/Transforms/Scalar.h"
 #include "llvm/Transforms/IPO/PassManagerBuilder.h"
